tests/LiquiditySwapV2.t.sol

```diff
@@ -0,0 +1,407 @@
+      true,
+      true
+    );
+
+    IERC20Detailed(collateralAssetAToken).approve(
+      address(liquiditySwapAdapter),
+      collateralAmountToSwap
+    );
+
+    IParaSwapLiquiditySwapAdapter.LiquiditySwapParams
+      memory liquiditySwapParams = IParaSwapLiquiditySwapAdapter.LiquiditySwapParams({
+        collateralAsset: collateralAsset,
+        collateralAmountToSwap: collateralAmountToSwap,
+        newCollateralAsset: newCollateralAsset,
+        newCollateralAmount: daiSupplyAmount,
+        offset: psp.offset,
+        user: user,
+        paraswapData: abi.encode(psp.swapCalldata, psp.augustus)
+      });
+
+    IParaSwapLiquiditySwapAdapter.FlashParams memory flashParams;
+    IParaSwapLiquiditySwapAdapter.PermitInput memory collateralATokenPermit;
+
+    liquiditySwapAdapter.swapLiquidity(liquiditySwapParams, flashParams, collateralATokenPermit);
+
+    uint256 daiCollateralAssetATokenBalanceAfter = IERC20Detailed(collateralAssetAToken).balanceOf(
+      user
+    );
+    assertTrue(
+      _withinRange(
+        daiCollateralAssetATokenBalanceBefore - daiCollateralAssetATokenBalanceAfter,
+        collateralAmountToSwap,
+        2
+      )
+    );
+    assertEq(daiCollateralAssetATokenBalanceAfter, 0);
+    _invariant(address(liquiditySwapAdapter), newCollateralAsset, newCollateralAssetAToken);
+    _invariant(address(liquiditySwapAdapter), collateralAsset, collateralAssetAToken);
+  }
+
+  function test_liquiditySwap_with_extra_collateral() public {
+    uint256 supplyAmount = 12000e18;
+    uint256 borrowAmount = 6000e18;
+    uint256 flashLoanAmount = 2000e18;
+    address collateralAssetAToken = AaveV2EthereumAssets.DAI_A_TOKEN;
+    address collateralAsset = AaveV2EthereumAssets.DAI_UNDERLYING;
+    address newCollateralAsset = AaveV2EthereumAssets.LUSD_UNDERLYING;
+    address newCollateralAssetAToken = AaveV2EthereumAssets.LUSD_A_TOKEN;
+    address flashLoanAsset = collateralAsset;
+    address flashLoanAssetAToken = collateralAssetAToken;
+    vm.startPrank(user);
+
+    _supply(AaveV2Ethereum.POOL, supplyAmount, collateralAsset);
+    _borrow(AaveV2Ethereum.POOL, borrowAmount, collateralAsset);
+
+    uint256 collateralAssetATokenBalanceBefore = IERC20Detailed(collateralAssetAToken).balanceOf(
+      user
+    );
+
+    uint256 collateralAmountToSwap = 2000e18;
+    PsPResponse memory psp = _fetchPSPRoute(
+      collateralAsset,
+      newCollateralAsset,
+      (collateralAmountToSwap * 9991)/10_000, //taking flashloan premium into account
+      user,
+      true,
+      false
+    );
+
+    IERC20Detailed(collateralAssetAToken).approve(
+      address(liquiditySwapAdapter),
+      collateralAmountToSwap
+    );
+
+    IParaSwapLiquiditySwapAdapter.LiquiditySwapParams
+      memory liquiditySwapParams = IParaSwapLiquiditySwapAdapter.LiquiditySwapParams({
+        collateralAsset: collateralAsset,
+        collateralAmountToSwap:(collateralAmountToSwap * 9991)/10_000, //taking flashloan premium into account
```

Comment:
flashloan premium is already taken into account by the contract right?
the adapter contract flashborrow assets and sells all (minus the fee).
It seems the contract is not using `liquiditySwapParams.collateralAmountToSwap` when using flashloans, right?

Reply:
yes. you are correct.